core/server/src/main/java/alluxio/master/file/meta/TempInodePathWithDescendant.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the “License”). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.file.meta;
+
+import alluxio.AlluxioURI;
+import alluxio.exception.FileDoesNotExistException;
+import alluxio.exception.InvalidPathException;
+
+import java.util.List;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * This class represents a temporary {@link InodePath}. This {@link InodePath} will not unlock the
+ * inodes on close. This {@link InodePath} can set any descendant (does not have to be immediate),
+ * useful for operations requiring an {@link InodePath}.
+ */
+@ThreadSafe
+public final class TempInodePathWithDescendant extends InodePath {
```

Comment:
do we plan to have `TempInodePathWithoutDescendant`? If not, probably just `TempInodePath`?